### PR TITLE
SC-7053 - added typescript dependencies and config

### PR DESCRIPTION
**Pull request by tofixx**
# Description
Introduce typescript

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -81,7 +81,6 @@
 	"jest.enableCodeLens": false,
 	"javascript.format.enable": false,
 	"json.format.enable": false,
-	"vetur.validation.script": false,
 	"vetur.format.defaultFormatter.js": "none",
 
 	// ===
```
Comment (tofixx) [inline .vscode/settings.json:86]: this is invalid json (as before...)

File: src/components/organisms/FormNews.unit.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ const getRouterPushSpy = (wrapper, expects) => {
 };
 
 describe("@components/organisms/FormNews", () => {
-	it(...isValidComponent(FormNews));
+	it.skip(...isValidComponent(FormNews));
```
Comment (tofixx) [inline src/components/organisms/FormNews.unit.js:73]: why this is skipped?

File: src/components/organisms/FormNews.vue [hunk 3]
```diff
@@ -181,19 +193,19 @@ export default {
 						this.$route.query.targetmodel || this.$route.query.context,
 				});
 				this.$toast.success(
-					this.$t("components.organisms.FormNews.success.create")
+					this.$t("components.organisms.FormNews.success.create").toString()
```
Comment (tofixx) [inline src/components/organisms/FormNews.vue:196]: can we have a typescripted this.$t which returns a string?

**Review by tofixx — COMMENTED** [review]
looks promising, what are the open challenges in the test and what is the result of the vue component variants, that let you choose Vue.extend?

File: src/components/organisms/FormNews.unit.js (hunk 2 above)
Comment (tofixx) [reply src/components/organisms/FormNews.unit.js:73]: this was checking the plain object 'type' only. typechecking will be done by typescript now and this is a function object now.--> can be removed.

File: src/components/organisms/FormNews.vue (hunk 3 above)
Comment (tofixx) [reply src/components/organisms/FormNews.vue:196]: `this.$ts("components.organisms.FormNews.success.create"):string`

File: src/components/organisms/FormNews.unit.js [hunk 4]
```diff
@@ -70,8 +70,6 @@ const getRouterPushSpy = (wrapper, expects) => {
 };
 
 describe("@components/organisms/FormNews", () => {
-	it.skip(...isValidComponent(FormNews));
```
Comment (haniapalac) [inline src/components/organisms/FormNews.unit.js]: Shouldn't all those tests be removed form the nuxt client?

**Review by pkleybolte — APPROVED** [review]

**Pull request merged**